lib/collection/src/shards/replica_set/partial_snapshot_meta.rs

```diff
@@ -40,4 +49,61 @@ impl PartialSnapshotMeta {
             })
         }
     }
+
+    pub fn take_read_lock(&self) -> CollectionResult<tokio::sync::OwnedSemaphorePermit> {
+        self.read_lock.clone().try_acquire_owned().map_err(|_| {
+            CollectionError::bad_request("partial snapshot recovery is already in progress")
+        })
+    }
+
+    pub fn recovery_timestamp(&self) -> u64 {
+        self.recovery_timestamp.load(atomic::Ordering::Relaxed)
+    }
+
+    pub fn snapshot_recovered(&self) {
+        let timestamp = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap_or(Duration::ZERO)
+            .as_secs();
+
+        self.recovery_timestamp
+            .store(timestamp, atomic::Ordering::Relaxed);
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct RequestTracker {
+    requests: Arc<AtomicUsize>,
```

Comment:
how it can be more than 1?

Reply:
`create_partial_snapshot` is **not** limited to one request at a time. Only `recover_from` is.

If you call `recover_from` from two different read nodes (with the same write node url), two create_partial_snapshot requests may be called on write node at the same time.

Reply:
See answer to the next comment.